### PR TITLE
Read Saint Kitts and Nevis' BUR1 from pdf

### PR DESCRIPTION
**Pull request by crdanielbusch**
## Description

Read Saint Kitts and Nevis BUR1 from PDF.

## Checklist

Please confirm that this pull request has done the following:

- [ ] Tests added
- [ ] Documentation added (where applicable)
- [ ] Changelog item added to `changelog/`

### CODE REVIEW

**Comment by JGuetschow** [comment]
I think we need the agriculture trend table from pages 131/132, so we can obtain all PRIMAP-hist sectors for all gases through downscaling.

**Comment by crdanielbusch** [comment]
> I think we need the agriculture trend table from pages 131/132, so we can obtain all PRIMAP-hist sectors for all gases through downscaling.

The data from the table on pages 131/132 is already read in with the overview table on pages 111/112/113.

We now have the sectors 3 / 3.A / 3.B / 3.C and 3.D  for 2008 to 2018 (with 3.D being NE for all years) and KYOTOGHG (AR6GWP100). Do you mean downscaling entities for the years 2008-2018? For example, create values for N2O and CH4 for 3.A for all the years based on the values of 2018?

**Comment by crdanielbusch** [comment]
@JGuetschow I updated the downscaling for most primap sectors. We now have the values by entity for 2008-2018  for the categories:

1, 1.A, 1.B, 3, 3.A, 3.B, 3.C, 3.D, 4

Still missing are the values for the categories 2, 2.A, 2.B, 2.C, 2.D, 2.E, 2.F, 2.G, 2.H. They are all zero and the downscaling currently does not work for that.

Also, we don't have the trend for 1.C, as there are no values for KYOTOGHG.

**Review by JGuetschow — CHANGES_REQUESTED** [review]

File: src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py [hunk 1]
```diff
@@ -0,0 +1,632 @@
...
+            # "1. Energy": "1. Energy",
+            "A. Fuel Combustion": "1.A",
+            "1.  Energy Industries": "1.A.1",
+            "2.  Man. Ind. & Constr.": "1.A.2",
+            "3.  Transport": "1.A.3",
+            "4.  Other Sectors": "1.A.4",
+            "5.  Other": "1.A.5",
+            "B. Fugitive Emissions from Fuels": "1.B",
+            "1.  Solid Fuels": "1.B.1",
+            "2.  Oil and Natural Gas and other…": "1.B.2",
```
[the remark this answers is not in the document]
Comment (crdanielbusch) [reply src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py:258]: The categories `1.B.2` and `1.B.3` are in the table on page 125. Wouldn't it be better to drop `M.1.B.23` and work with the values directly assigned to the categories?

File: src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py [hunk 2]
```diff
@@ -0,0 +1,632 @@
...
+            "D.  Non-energy products": "2.D",
+            "E.  Electronics industry": "2.E",
+            "F.  Product uses as ODS substitutes": "2.F",
+            "G.  Other product manufacture and": "2.G",
+            "use  H.  Other": "2.H",
+            "3.  Agriculture": "M.AG",
+            "A.  Enteric Fermentation": "3.A.1",
+            "B.  Manure Management": "3.A.2",
+            "C.  Rice Cultivation": "3.C.7",
+            "D.  Agricultural Soils": "3.C.4",  # TODO confirm! 3c4 and 3c5
```
[the remark this answers is not in the document]
Comment (crdanielbusch) [reply src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py:272]: Ok noted

File: src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py [hunk 3]
```diff
@@ -0,0 +1,632 @@
...
+    "KYOTOGHG (SARGWP100)": ["CO2", "CH4", "N2O", "FGASES (SARGWP100)"],
+    "KYOTOGHG (AR4GWP100)": ["CO2", "CH4", "N2O", "FGASES (AR4GWP100)"],
+    "KYOTOGHG (AR5GWP100)": ["CO2", "CH4", "N2O", "FGASES (AR5GWP100)"],
+    "KYOTOGHG (AR6GWP100)": ["CO2", "CH4", "N2O", "FGASES (AR6GWP100)"],
+}
+
+country_processing_step1 = {
+    "tolerance": 0.01,
+    "aggregate_cats": {
+        "M.3.D.AG": {"sources": ["3.D.2"]},
```
[the remark this answers is not in the document]
Comment (crdanielbusch) [reply src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py]: Haven't seen this and already changed it in the code

File: src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py [hunk 4]
```diff
@@ -0,0 +1,632 @@
...
+                "3.C.5",
+                "3.C.6",
+                "3.C.7",
+                "3.C.8",
+            ]
+        },
+        "3.D": {"sources": ["3.D.1", "3.D.2"]},
+        "M.AG": {"sources": ["3.A", "M.AG.ELV"]},
+        "M.3.D.LU": {"sources": ["3.D.1"]},
+        "M.LULUCF": {"sources": ["3.B", "M.3.D.LU"]},
```
[the remark this answers is not in the document]
Comment (crdanielbusch) [reply src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py]: Done

File: src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py [hunk 5]
```diff
@@ -0,0 +1,632 @@
...
+            "3.  Agriculture": "M.AG",
+            "A.  Enteric Fermentation": "3.A.1",
+            "B.  Manure Management": "3.A.2",
+            "C.  Rice Cultivation": "3.C.7",
+            "D.  Agricultural Soils": "3.C.4",  # TODO confirm! 3c4 and 3c5
+            "E.  Prescribed Burning of Savannahs": "3.C.1.d",  # TODO confirm!
+            "F.  Field Burning of Agricultural": "3.C.1.b",  # TODO confirm!
+            "Residues  G.  Liming": "3.C.2",
+            "H.  Urea applications": "3.C.3",
+            "I.  Other carbon-containing": "3.D.2",  # TODO confirm! ?
```
[the remark this answers is not in the document]
Comment (crdanielbusch) [reply src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py]: Done

File: src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py [hunk 6]
```diff
@@ -0,0 +1,632 @@
...
+            "Information Items": "MEMO",
+            "CO2 from Biomass Combustion for Energy Production": "MBIO",
+        },
+        "page_defs": {
+            "153": {"skip_rows_start": 2},
+            "154": {"skip_rows_start": 2},
+            "155": {"skip_rows_start": 2},
+        },
+        "replace_data_entries": {
+            "NO,NE": "NO",
```
[the remark this answers is not in the document]
Comment (crdanielbusch) [reply src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py:385]: Ok I can do that

File: src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JGuetschow) [reply src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py]: If they are present already, that's of course better than downscaling. Though I see only aggregate Kyoto GHG in the table, so some downscaling is still needed.

File: src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (crdanielbusch) [reply src/unfccc_ghg_data/unfccc_reader/Saint_Kitts_and_Nevis/config_kna_bur1.py:385]: Had a closer look at this one. I don't think it's necessary to open an issue. I replaced the `,` with `.` in the data column. The values `NO,NE` then turned into `NO.NE` which can recognised as string code anymore.

**Review by JGuetschow — APPROVED** [review]

**Pull request merged**